packages/lit-dev-content/site/tutorials/content/reactivity/04.md

```diff
@@ -0,0 +1,9 @@
+One way your component can react to property changes is to compute values based on the new and existing property values.
+
+If you need to compute a value to render, you can compute it in the `render()` method itself. However a more general solution is to compute values in the `willUpdate()` callback. This has a couple of advantages:
+
+-   You can make computed properties available to [reactive controllers](/docs/composition/controllers/). The `willUpdate()` callback is called before the `hostUpdate()` callback is called on any [reactive controllers](/docs/composition/controllers/).
```

Comment:
This is super advanced and could easily be removed.

Reply:
Yeah, I basically included this to explain the advantages of `willUpdate` over `render` ... I was going to mention that you got `changedProperties` as well, but I think that you have a change that exposes `changedProperties` on the component, which weakens that argument...

Anyway--if we just present `willUpdate` as the default way to do this, we don't need to explain the advantages.